### PR TITLE
Deeper learning: Remove deprecated UI

### PR DESCRIPTION
**Pull request by islemaster**
Two UI changes to the peer review dashboard. ([PLC-67](https://codedotorg.atlassian.net/browse/PLC-67), [spec](https://docs.google.com/document/d/1rtSHDfNh2E6sVpoF1kM2zIy0SQRHrfX3bfnkp5AHEW8/edit))

1. Remove the "Escalated" and "Open" tabs.
   Actually, remove the tabs entirely since we only care about the "All Submissions" view now.  I've also removed this filter option from the API, since it's now unused.

2. Remove the "Accepted submissions" and "Rejected submissions" columns.
  These columns aren't useful in the workflow this year and they take up quite a bit of space.  I've also removed these counts from the summary data returned by the API.

Before:
before

After:
after

Done now, because it was fairly easy to do this cleanup while investigating and fixing a different Deeper Learning issue, in https://github.com/code-dot-org/code-dot-org/pull/27406.

### CODE REVIEW

File: apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx [hunk 1]
```diff
@@ -11,7 +11,6 @@ export default storybook => {
         name: 'Escalated Peer Review Submissions',
```
Comment (clareconstantine) [inline apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx]: How are these stories different now, if both have the escalated status for all the submissions?

File: apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx [hunk 2]
```diff
@@ -61,9 +55,7 @@ export default storybook => {
                 submission_date: '2/19/2017',
                 escalation_date: undefined,
                 review_ids: [[1, ''], [2, 'accepted']],
-                status: 'escalated',
-                accepted_reviews: 1,
-                rejected_reviews: 0
+                status: 'escalated'
```
Comment (clareconstantine) [inline apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx]: How do we get in a state where the `status` is 'escalated' but `escalation_date` is undefined?

**Review by clareconstantine — APPROVED** [review]

File: apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx (hunk 1 above)
Comment (islemaster) [reply apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx]: 🤦‍♂️ Great catch - they're not! I'll tear one out.

File: apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx (hunk 2 above)
Comment (islemaster) [reply apps/src/code-studio/peer_reviews/PeerReviewSubmissionData.story.jsx]: I'm not sure - this may not be a representative story.  I'll review.

**Pull request merged**